libminifi/test/flow-tests/FlowControllerTests.cpp

```diff
@@ -133,12 +144,9 @@ TEST_CASE("Flow shutdown waits for a while", "[TestFlow2]") {
   testController.startFlow();
 
   // wait for the source processor to enqueue its flowFiles
-  int tryCount = 0;
-  while (tryCount++ < 10 && root->getTotalFlowFileCount() != 3) {
-    std::this_thread::sleep_for(std::chrono::milliseconds{20});
-  }
+  busy_wait(std::chrono::milliseconds{50}, [&] {return root->getTotalFlowFileCount() != 0;});
```

Comment:
Why not simply:
```c++
assert(verifyEventHappenedInPollTime(std::chrono::milliseconds{50}, [&] {return root->getTotalFlowFileCount() != 0;})
```

Reply:
The same applies to the other similar changes.

Reply:
see [this](https://github.com/apache/nifi-minifi-cpp/pull/861#discussion_r465030783), we could extend the `verifyEventHappenedInPollTime` with a `poll_interval_ms` where `-1` would indicate a busy checking, what do you think?

Reply:
I would prefer a function without a timeout parameter (possibly with both the timeout-y and timeout-less version calling a common implementation).

Reply:
I cannot see where to link points, but -1 from me for the additional argument as deducing if we need to sleep should be a compile-time operation.

Reply:
@hunyadi-dev I don't see why it should be compile-time decidable if we need sleep but alright will not add an extra parameter, @fgerlits could you expand on your proposal?

Reply:
@adebreceni It wasn't really a proposal, but I would prefer two separate functions `verifyEventHappenedInPollTime()` and `verifyEventHappenedInPollTimeWithBusyWait()` (just an example, feel free to find a shorter name) rather than a single function where an argument value of -1 means "use the busy version".

[TBH, I am not a fan of the `verifyEventHappenedInPollTime` name, as something like `REQUIRE(happensBeforeTimeout(...))` would read better, but renaming it is probably not in scope for this PR.]

Reply:
changed it to `verifyWithBusyWait`